### ISSUE TITLE
Various roles of energy are not very clear

### ISSUE THREAD

**Issue by Muon** [ZeroK-RTS/Zero-K#548]
The role of energy in Zero-K is fairly complicated, and some of it is not super obvious. Off the top of my head:
- NOT used to power mexes: mexes don't consume any power to get the base rate of their mexspot.
  - I only noticed this recently; I tacitly assumed that mexes drained energy, because people always built power immediately after mexes, and that was why I needed to provide them with solars, rather than because buildpower drains energy.
- Used to power radar.
  - And also certain other buildings, but I can't remember which.
- Used to build things (cf. #396).
- Used to overdrive mexes within the same energy grid when...?
  - I used to think it was when energy would be wasted, but then I looked at the energy consumption breakdown, and even though I wasn't at capacity, there was some overdrive usage.
- Used to power Aspis shield, but **not** Thug.
- Used for cloak and resurrection.
- Builders and caretakers also generate a tiny amount of income.

UI improvements could probably help clarify, but there are lots of little cases to remember. Simplifying some of it would likely be best.

**Comment by Skasi** [comment]
+1

There's many different things to know. Not all of them should be removed, as that would probably dumb down the game too much and take away strategy (eg. kill energy to disable enemy radars/cloak/shields). However, some simplifications and visualization of effects could definitely help.

Shield and cloak energy consumption feels very "random". A similar lack of clarity exists for decloak ranges.

Overdrive is already shown graphically by fast mex animations. I suggested using mex glow for overdrive as well. Right now orange mex glow is based off mex output, not overdrive.

Things like repair- and resurrection energy cost- and buildpower multipliers could probably benefit from graphical effects as well, but that requires brainstorming for ideas.

**Comment by Anarchid** [comment]
"drains energy" icon to aspis/radar/cloakers tooltip?

**Comment by Muon** [comment]
@Skasi I wasn't really concerned about highlighting overdrive, but that it wasn't clear how and when energy is used to overdrive mexes. I'm still not clear on how overdrive is applied, which is kinda annoying.

I agree that the shield and cloak energy numbers feel inconsistent. I feel that I should be able to say off the top of my head exactly how much cloak costs after having played 154 games, but I really can't.

@Anarchid Sounds like a step in the right direction. That said, I feel like the amount of energy drained ought to be shown in the tooltip, since the quantities aren't obvious or constant.

Perhaps exposing the current used and total available buildpower somewhere in the economy UI would be useful?

**Comment by Anarchid** [comment]
> Perhaps exposing the current used and total available buildpower somewhere in the economy UI would be useful?

This already happens. Evidently, it is insufficient.

**Comment by sprunk** [comment]
Half of the ticket could be closed since most of the stuff is not changeable, energy is supposed to have many uses. 

Some things could reasonably be changed. The existence of rez and the metalmaker sidearm (ie +0.3 con incomes) would be such design decisions (though unlikely to pass, was tried already).
.

**Comment by GoogleFrog** [comment]
I don't think there is anything more to be done. I think we now have a good level of discoverability relative to how important it is that each facet of energy be discovered.

There are basically two important parts of energy:
 * Construction costs energy.
 * Large cloak/shield as well as some cloak costs significant energy.

Construction costs are hopefully covered by the merged metal+energy icon in the tooltip. Large energy drains are hard to come across and will often be from something the player is paying particular attention to which makes them more likely to notice the little display which shows energy drain.

New players are not going to notice minor energy incomes or drains such as the income of a constructor or the drain of a radar. Cloak, shield and misc costs are listed in the space+click window which is about as discoverable as they need to be. Furthermore the UI always shows the energy income and expenditure of the units you have selected. You're not going to remember the exact costs but that is unimportant, you just need a feel for what is expensive or cheap. New players shouldn't be distracted by energy sinks unless they are large enough to destroy their economy. More experienced players can delve deeper with the wiki or ingame help, it is by no means hidden so I'm just caring about what is required for a new players. Whether or not mexes drain energy is likewise unimportant.

Overdrive is not important to teach. There is little lost if a new player does not even realize that overdrive exists. They can learn about it later.

The disparate energy drains for cloaking and shields are designed with good reason, not randomly assigned. An energy cost effectively buffs a unit lategame while nerfing it in the early game. There is not going to be any wide-reaching drain consistency, although there could be some consistency within a few classes.

TLDR: It's complicated if you really want to dig into it and learn exactly how to optimize your energy. Otherwise it is very simple, make enough energy to spend your metal and runs you large cloakers and shields.
